docker/deploy.sh

```diff
@@ -78,6 +86,13 @@ run_docker_swarm() {
             sleep 1
             exit 1
         fi
+        if ! docker stack deploy -c $docker_file -c $gw_docker_file op-scim
+        then
+            echo " "
+            echo "Failed to deploy to Docker Swarm; investigate the error before proceeding"
+            sleep 1
+            exit 1
+        fi
     fi
 
     if ! docker stack deploy -c $docker_file op-scim
```

Comment:
I wonder if this is necessary? Is this not covered by the `docker stack deploy` on line 67? Looks like it _may_ be a holdover from before your refactor.

Reply:
Line 67 applies the standard docker-compose.yml from the swarm directory if the deployer had selected no for using Google Workspace, whereas line 89 would apply if the deployer had selected yes for using Google Workspace and apply both the docker-compose.yml and the gw-docker-compose.yml from the swarm directory with the settings from gw-docker-compose.yml ontop of the docker-compose.yml.

Reply:
interesting, my comment may not have attached itself to the correct line. I was thinking of the `if` conditional beginning on line 98. The `if ! workspaceIDP` on line 65 takes care of both conditions, now? If not workspace, deploy using the docker-compose, otherwise deploy using workspace-docker-compose. So that conditional takes care of all possible conditions. 

But line 98 is `another docker stack deploy -c $docker_file` that is not part of any other conditional. Isn't this already covered by TRUE condition of line 65? (If NOT workspace)

Reply:
Correct, on line 98 it is a holdover from before, I will remove it. Nice catch!